docs/api-testing/mark3/mocking.md

```diff
@@ -0,0 +1,271 @@
+
+Why the name Piestry?
+didn't want to drop the Sauce tradition of naming stuff after food, so Piestry is:
+"A pastry that acts as if it was a pie, but is nothing more than a pastry"
+
+
+--use cases: you want to fake the payment transaction, you want to isolate the microservice from the rest of the API actions (ie everything else is stable and you want to drill down to find the error)
+
+
+Where can I run it?
+* Docker container (locally or CI/CI pipeline)
+  * Image: quay.io/saucelabs/piestry
+  * Run it using the code snippet below:
+  ```bash
+  docker run -v "$(pwd)/specs:/specs" -p 5000:5000 quay.io/saucelabs/piestry -u /specs/myspec.yaml
+  Where /specs/myspec.yaml is the URI to a YAML file, local or remote.
+  ```
+
+## What You'll Need
+
+* A Sauce Labs account ([Log in](https://accounts.saucelabs.com/am/XUI/#login/) or sign up for a [free trial license](https://saucelabs.com/sign-up))
+* An OpenAPI spec file
+
+
+## Mock Generation
+
+If you provide a regular OpenAPI spec file, the system should bind a series of endpoints to simulate whatever is in the spec.
+
+* When only a response schema is present: the system will generate random data for each field.
+* When one response example is present: the system will present the example.
+* When multiple response examples are present: the system will present the first example.
+* When multiple content types are available: the system will pick the one closer to the "Accept" header, any JSON response if a match is not found.
+
+1. Place your spec file (or set of files in a folder) in a location of your choice. For this example, we'll call it `myspec.yaml`.
+2. Open your CLI terminal and navigate to right outside that folder, then run this command in your terminal:
+  ```bash
+  docker run -v "$(pwd)/myspec:/specs" -p 5000:5000 quay.io/saucelabs/piestry -u /specs/myspec.yaml
+  ```
+  `$(pwd)/myspec` means the `{current_directory}/myspec` that gets mounted to the container in the `/specs` folder. Therefore, the -u (relative to the container is) `/specs/myspec.yaml`.
+3. If all goes well, you should see the listing of the available routes:
+  ```json
+  2021-10-05T07:32:35.157Z info: Piestry booting on port: 5000
+  2021-10-05T07:32:35.189Z info: Registering GET /api/v1/release-notes
+  2021-10-05T07:32:35.191Z info: Registering GET /api/v1/user
+  2021-10-05T07:32:35.191Z info: Registering GET /api/v1/user/:id
+  2021-10-05T07:32:35.192Z info: Registering GET /api/v1/echo
+  2021-10-05T07:32:35.192Z info: Registering POST /api/v1/echo
+  2021-10-05T07:32:35.192Z info: Registering POST /api/v1/post-check
+  2021-10-05T07:32:35.193Z info: Registering POST /api/v1/check-in
+  ```
+4. At this point, you can use any **HTTP client** to query one of these endpoints (i.e., `curl localhost:5000/api/v1/release-notes`). It should return a mock for release notes. On top of this, you can add the option to connect the logger, and there you go.
```

Comment:
4. At this point, you can use any **HTTP client** to query one of these endpoints (i.e., `curl localhost:5000/api/v1/release-notes`: it should return a mock for release notes).